File: course1/week2/4_least_common_multiple/java/src/Main.java
/**
 *
 * Java implementation to find the least common multiple between two numbers
 *
 * (c) Copyright 2019 Clayton J. Wong ( http://www.claytonjwong.com )
 *
 **/

import java.util.Scanner;

public class Main {

    private static long gcd( long a, long b ){
        for( long t = 0; b != 0; ){
            t = b;
            b = a % b;
            a = t;
        }
        return a;
    }

    private static long lcm( long a, long b ){
        return a * b / gcd( a, b );
    }

    public static void main(String[] args){
        Scanner input = new Scanner( System.in );
        long a = input.nextLong(),
             b = input.nextLong(),
             ans = lcm( a, b );
        System.out.println( ans );
    }
}

File: course1/week2/4_least_common_multiple/cpp/main.cpp
/**
 *
 * C++ implementation to calculate the least common multiple between two numbers
 *
 * (c) Copyright 2019 Clayton J. Wong ( http://www.claytonjwong.com )
 *
 **/

#include <iostream>

using namespace std;

template< typename Type >
class Solution {
public:

    Type gcd( Type a, Type b ){
        for( auto t{ b }; b != 0; ){
            t = b;
            b = a % b;
            a = t;
        }
        return a;
    }

    Type lcm( Type a, Type b ){
        return a * b / gcd( a, b );
    }
};

int main(){
    using Type = size_t;
    Solution< Type > solution;
    Type a = 0,
        b = 0;
    cin >> a >> b;
    Type ans = solution.lcm( a, b );
    cout << ans << endl;
    return 0;
}
